scripts/analysis.py

```diff
@@ -0,0 +1,128 @@
+    """
+    send_requests = pd.json_normalize(
+        [event for event in trace_events if event['name'] == 'ResourceSendRequest']
+    )
+    finish_requests = pd.json_normalize(
+        [event for event in trace_events if event['name'] == 'ResourceFinish']
+    )
+    data = send_requests.merge(
+        finish_requests, on='args.data.requestId', how='left', suffixes=('_send', '_finish')
+    )[
+        [
+            'ts_send',
+            'ts_finish',
+            'args.data.encodedDataLength',
+            'args.data.priority',
+            'args.data.url',
+            'args.data.requestMethod',
+        ]
+    ].rename(
+        {
+            'ts_send': 'request_start',
+            'ts_finish': 'response_end',
+            'args.data.encodedDataLength': 'encoded_data_length',
+            'args.data.url': 'url',
+            'args.data.requestMethod': 'method',
+        },
+        axis=1,
+    )
+    data['request_start'] = data['request_start'] * 1e-3
+    data['response_end'] = data['response_end'] * 1e-3
+    data['total_response_time_ms'] = data['response_end'] - data['request_start']
+    if url_filter:
+        data = data[data['url'].str.contains(url_filter)]
+    return data
+
+
+def load_frame_data(*, frame_data_path: str):
+    """
+    Load frame data from the output of Chrome dev tools.
+
+    Parameters
+    ----------
+
+    frame_data_path: str
+        Path to the json containing frame data
+
+    Returns
+    -------
+    frame_data : pd:DataFrame
+        DataFrame containing frame timing and durations
+    """
+    with open(frame_data_path) as f:
+        frames = json.load(f)
+    frames = pd.json_normalize(frames).sort_values(by='startTime')[
+        ['startTime', 'endTime', 'duration']
+    ]
+    return frames
+
+
+def plot_requests(df: pd.DataFrame):
+    """
+    Plot rectangles showing each request duration
+    """
+    df['rectangle'] = df.apply(
+        lambda x: (x['request_start'], x.name, x['response_end'], x.name + 1), axis=1
+    )
+    boxes = hv.Rectangles(df['rectangle'].to_list())
+    boxes.opts(width=1000, color='lightgrey', xlabel='Timestamp', yaxis=None)
+    return boxes
+
+
+def plot_frames(df: pd.DataFrame):
+    """
+    Plot rectangles showing each fra,e duration
+    """
+    df['rectangle'] = df.apply(lambda x: (x['startTime'], 1, x['endTime'], 2), axis=1)
+    boxes = hv.Rectangles(df['rectangle'].to_list())
+    boxes.opts(width=1000, color='lightgreen', xlabel='Timestamp', yaxis=None)
+    return boxes
+
+
+# Parse command line arguments and run main function
+if __name__ == '__main__':
+    # Parse input args
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--timestamp', type=str)
+    parser.add_argument('--run', type=int, default=1)
+    args = parser.parse_args()
+    trace_data_fp = f'chrome-devtools-traces/{args.timestamp}-{args.run}.json'
+    frame_data_fp = f'chrome-devtools-traces/frames/{args.timestamp}-{args.run}.json'
```

Comment:
@maxrjones, i'm curious... how is `chrome-devetools-traces/frames` directory populated? i tried running the following, 

```bash
❯ python scripts/analysis.py --timestamp 2023-07-26T18-53-10 --run 2
Traceback (most recent call last):
  File "/home/andersy005/devel/carbonplan/benchmark-maps/scripts/analysis.py", line 124, in <module>
    frame_data = load_frame_data(frame_data_path=frame_data_fp)
  File "/home/andersy005/devel/carbonplan/benchmark-maps/scripts/analysis.py", line 77, in load_frame_data
    with open(frame_data_path) as f:
FileNotFoundError: [Errno 2] No such file or directory: 'chrome-devtools-traces/frames/2023-07-26T18-53-10-2.json'

```

Reply:
I generated that file manually using your screen recording guide. I will open a PR with one of these data files for experimentation, along with another follow-up commit from this branch.